### ISSUE TITLE
How can we handle resource exhaust error coming while allocating a tensor of a particular size.

### ISSUE THREAD

**Issue by alok2490** [tensorflow/tensorflow#18733]
I am currently using single GPU : GEFORCE GTX1080 8GB and I am unable to handle an error coming while I am trying to train a network. Error is :

ResourceExhaustedError (see above for traceback): OOM when allocating tensor with shape[131072,2048]
[[Node: training/Adam/mul_3 = Mul[T=DT_FLOAT, _device="/job:localhost/replica:0/task:0/gpu:0"](Adam/beta_1/read, training/Adam/Variable/read)]]
[[Node: loss/mul/_179 = _Recvclient_terminated=false, recv_device="/job:localhost/replica:0/task:0/cpu:0", send_device="/job:localhost/replica:0/task:0/gpu:0", send_device_incarnation=1, tensor_name="edge_697_loss/mul", tensor_type=DT_FLOAT, _device="/job:localhost/replica:0/task:0/cpu:0"]]

However, I understand that I have limited memory in GPU which has to be utilized. I currently just wanting to know if there is any workaround of this by which I can train the network without reducing the dimensions of the network.

Model :

    model = Sequential()
    model.add(TimeDistributed(Flatten(), input_shape = (24,8,8,2048)))
    model.add(Dropout(0.5))
    model.add(LSTM(512, return_sequences=True, dropout=0.5))
    model.add(LSTM(512, return_sequences=False, dropout=0.5))
    model.add(Dense(512, activation='relu'))
    model.add(Dropout(0.5))
    model.add(Dense(256, activation='relu'))
    model.add(Dropout(0.5))
    model.add(Dense(5, activation='softmax'))
Input to this network is actually image embedding I am getting from InceptionV3 without top.

Any help in this regard is appreciated.

Thanks in advance.

**Comment by tensorflowbutler** [comment]
Thank you for your post. We noticed you have not filled out the following field in the issue template. Could you update them if they are relevant in your case, or leave them as N/A? Thanks.
Have I written custom code
OS Platform and Distribution
TensorFlow installed from
TensorFlow version
Bazel version
CUDA/cuDNN version
GPU model and memory
Exact command to reproduce

**Comment by Baschdl** [comment]
It's a huge tensor you try to allocate. If we assume that you use a data type which needs 32bit for each value, your complete tensor needs 131072*2048*32bit/1024/1024/1024=8GB of memory on your GPU. Can you reduce some dimension like the embedding size or the time steps?

**Comment by michaelisard** [comment]
This question is better asked on [StackOverflow](http://stackoverflow.com/questions/tagged/tensorflow) since it is not a bug or feature request. There is also a larger community that reads questions there.

If you think we've misinterpreted a bug, please comment again with a clear explanation, as well as all of the information requested in the [issue template](https://github.com/tensorflow/tensorflow/issues/new). Thanks!
